wpimath/src/main/native/include/frc/MathUtil.h

```diff
@@ -169,4 +169,38 @@ constexpr units::radian_t AngleModulus(units::radian_t angle) {
                                        units::radian_t{std::numbers::pi});
 }
 
+// floorDiv and floorMod algorithms taken from Java
+
+/**
+ * Returns the largest (closest to positive infinity)
+ * {@code int} value that is less than or equal to the algebraic quotient.
+ *
+ * @param x the dividend
+ * @param y the divisor
+ * @return the largest (closest to positive infinity)
+ *   {@code int} value that is less than or equal to the algebraic quotient.
+ */
+constexpr static int FloorDiv(int x, int y) {
+  int r = x / y;
+  // if the signs are different and modulo not zero, round down
+  if ((x ^ y) < 0 && (r * y != x)) {
+    r--;
+  }
+  return r;
+}
```

Comment:
```suggestion
constexpr std::integral auto FloorDiv(std::integral auto x, std::integral auto y) {
  auto quot = x / y;
  auto rem = x % y;
  // if the signs are different and modulo not zero, round down
  if (x ^ y < 0 && rem != 0) {
    --quot;
  }
  return quot;
}
```
At least on x86, the quotient and remainder are computed in one idiv instruction, so you get the remainder for free.